sailfishusermanager.cpp

```diff
@@ -348,3 +349,65 @@ bool SailfishUserManager::hasAccessRights(uint uid_to_modify) {
 
     return true;
 }
+
+void SailfishUserManager::userSwitch(uint uid)
+{
+    if (!hasAccessRights(SAILFISH_UNDEFINED_UID)) {
+        auto message = QStringLiteral("Access denied");
+        qCWarning(lcSUM) << message;
+        sendErrorReply(QDBusError::AccessDenied, message);
+        return;
+    }
+
+    uid_t currentUid;
+    if (sd_seat_get_active("seat0", nullptr, &currentUid) < 0) {
+        auto message = QStringLiteral("Failed to get current user id");
+        qCWarning(lcSUM) << message;
+        sendErrorReply(QDBusError::Failed, message);
+        return;
+    }
+
+    QDBusInterface systemd(QStringLiteral("org.freedesktop.systemd1"), QStringLiteral("/org/freedesktop/systemd1"),
+                           QStringLiteral("org.freedesktop.systemd1.Manager"), QDBusConnection::systemBus());
```

Comment:
How about defining all string constants (especially those used repeatedly) somewhere in one place? As static objects, which might even reduce the size of the readonly-only data segment.

Reply:
Done